Drop-In/src/main/java/com/braintreepayments/api/DropInActivity.java

```diff
@@ -329,15 +334,21 @@ private void showBottomSheet() {
 
     private void showCardDetailsFragment(final String cardNumber) {
         if (shouldAddFragment(CARD_DETAILS_TAG)) {
-            dropInClient.getConfiguration((configuration, error) -> {
-                if (configuration != null) {
-                    // TODO: implement dropInClient.hasAuthType(AuthType.TOKENIZATION_KEY)
-                    boolean hasTokenizationKeyAuth =
-                            Authorization.isTokenizationKey(dropInClient.getAuthorization().toString());
-
-                    CardDetailsFragment cardDetailsFragment = CardDetailsFragment.from(
-                            dropInRequest, cardNumber, configuration, hasTokenizationKeyAuth);
-                    replaceExistingFragment(cardDetailsFragment, CARD_DETAILS_TAG);
+            dropInClient.getAuthorization(new AuthorizationCallback() {
+                @Override
+                public void onAuthorizationResult(@Nullable Authorization authorization, @Nullable Exception authorizationError) {
+                    dropInClient.getConfiguration((configuration, configurationError) -> {
+                        if (configuration != null) {
+                            boolean hasTokenizationKeyAuth =
+                                    Authorization.isTokenizationKey(authorization.toString());
+
+                            CardDetailsFragment cardDetailsFragment = CardDetailsFragment.from(
+                                    dropInRequest, cardNumber, configuration, hasTokenizationKeyAuth);
+                            replaceExistingFragment(cardDetailsFragment, CARD_DETAILS_TAG);
+                        } else {
+                            finishDropInWithError(authorizationError);
```

Comment:
```suggestion
                            finishDropInWithError(configurationError);
```
I think this is for handling the config error, but we should also handle the auth error

Reply:
Ah yeah good call. I'll update.